src/main/java/org/apache/commons/compress/compressors/CompressorStreamFactory.java

```diff
@@ -226,10 +227,31 @@ private static Iterable<CompressorStreamProvider> archiveStreamProviderIterable(
      * @since 1.14
      */
     public static String detect(final InputStream inputStream) throws CompressorException {
+        final Set<String> defaultCompressorNamesForDetection =
+                Sets.newHashSet(BZIP2, GZIP, PACK200, SNAPPY_FRAMED, Z, DEFLATE, XZ, LZMA, LZ4_FRAMED, ZSTANDARD);
+        return detect(inputStream, defaultCompressorNamesForDetection);
+    }
+    /**
+     * Try to detect the type of compressor stream while limiting the type to the provided set of compressor names.
+     *
+     * @param inputStream input stream
+     * @param compressorNames compressor names to limit autodetection
+     * @return type of compressor stream detected
+     * @throws CompressorException if no compressor stream type was detected
+     *                             or if something else went wrong
+     * @throws IllegalArgumentException if stream is null or does not support mark
+     *
+     * @since 1.24
+     */
+    public static String detect(final InputStream inputStream, final Set<String> compressorNames) throws CompressorException {
```

Comment:
Make package-private, no need to grow the public/protected API footprint.

Reply:
Up to you, as long as the _new_ code is 100% covered (or as as close to 100% as possible).